bazel/envoy_internal.bzl

```diff
@@ -27,12 +27,13 @@ def envoy_copts(repository, test = False):
         "-Zc:__cplusplus",
         "-std:c++14",
         "-DWIN32",
-        "-D_WIN32_WINNT=0x0602",
-        "-DNTDDI_VERSION=0x06020000",
+        "-D_WIN32_WINNT=0x0A00",
```

Comment:
Can we have a comment on this? "Updating" to windows 10? https://docs.microsoft.com/en-us/cpp/porting/modifying-winver-and-win32-winnt?view=vs-2019

Reply:
We could comment it, although we never had a (supported) version of windows before. This doesn't lock us into using Windows Server 2016/Windows 10 API's, it only makes them accessible without further changes to our build flags. Server 2012 is beyond EOL, so this is a simple switch to "enable all supported API's"

Reply:
I don't mean this is locking us into Windows 10API, just add a comment because `0x0A00` is cryptic.

Reply:
Clarified both the top comment, and added #defined names of these values.